### PR TITLE
Use OIDC (trusted publishers) for PyPI uploads

### PR DESCRIPTION
**Pull request by arkid15r**
## Proposed change

Use OIDC (trusted publishers) for PyPI uploads. This approach provides publishing ability with no stored permanent tokens.
Additional changes:
  - Reformat configuration file
  - Add pypy-3.9, pypy-3.10 to test matrix

## Type of change

- [ ] New country/market holidays support (thank you!)
- [ ] Supported country/market holidays update (calendar discrepancy fix, localization)
- [x] Existing code/documentation/test/process quality improvement (best practice, cleanup, refactoring, optimization)
- [ ] Dependency update (version deprecation/upgrade)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (a code change causing existing functionality to break)
- [ ] New feature (new python-holidays functionality in general)

## Checklist

- [x] I've followed the [contributing guidelines][contributing-guidelines]
- [x] I've added references to all holidays information sources used in this PR
- [x] The code style looks good: `make pre-commit` command generates no changes
- [x] All tests pass locally: `make test`, `make tox` (we strongly encourage adding tests to your code)

[contributing-guidelines]: https://github.com/vacanza/python-holidays/blob/beta/CONTRIBUTING.rst
[docs]: https://github.com/vacanza/python-holidays/tree/beta/docs/source

### CODE REVIEW

File: .github/workflows/ci-cd.yml [hunk 1]
```diff
@@ -1,35 +1,52 @@
...
+          - macOS-latest
+          - ubuntu-latest
+          - windows-latest
+        python-version:
+          - '3.8'
+          - '3.9'
+          - '3.10'
+          - '3.11'
+          - '3.12'
+          - 'pypy-3.9'
```
Comment (KJhellico) [inline .github/workflows/ci-cd.yml:44]: `pypy-3.8` removed?

File: .github/workflows/ci-cd.yml [hunk 2]
```diff
@@ -1,35 +1,52 @@
-name: Tests
-on: [merge_group, push, pull_request, workflow_dispatch]
+name: CI/CD
```
Comment (KJhellico) [inline .github/workflows/ci-cd.yml:1]: We need to change link in README as well:
```diff
-    .. image:: https://github.com/vacanza/python-holidays/workflows/Tests/badge.svg
+    .. image:: https://github.com/vacanza/python-holidays/workflows/CI%2FCD/badge.svg
```

Comment (arkid15r) [reply .github/workflows/ci-cd.yml:1]: Yep, missed this one. Updated.

File: .github/workflows/ci-cd.yml (hunk 1 above)
Comment (arkid15r) [reply .github/workflows/ci-cd.yml:44]: Whereas this one was intentional -- I looked at https://www.pypy.org/download.html and it seems 3.8 is not in the latest package version. I don't mind keeping it though if anyone suggests it's worth it.

**Pull request merged**